src/components/dapp-info.js

```diff
@@ -1,39 +1,134 @@
+    for (const [name, contract] of Object.entries(origin.contractService.marketplaceContracts)) {
+      const contractObj = await origin.contractService.deployed(contract)
+      data.push(<tr><td>{name} address</td><td>{contractObj._address}</td></tr>)
+      data.push(<tr><td>{name} ABI </td><td>{JSON.stringify(contract.abi)}</td></tr>)
+    }
+
+    const tokenContractName = 'OriginToken'
+    const tokenContract = origin.contractService.contracts[tokenContractName]
+    const token = await origin.contractService.deployed(tokenContract)
+    data.push(<tr><td>{tokenContractName} address </td><td>{token._address}</td></tr>)
+    data.push(<tr><td>{tokenContractName} ABI </td><td>{JSON.stringify(tokenContract.abi)}</td></tr>)
+    return data
+  }
+
+  async componentDidMount() {
+    this.loadMarketplaceData()
+      .then(res => this.setState(Object.assign(this.state, { marketplaceData: res })))
+      .catch(e => this.setState(Object.assign(this.state, {
+        marketplaceData: `error fetching marketplace data: ${e.message}`
+      })))
+    this.loadTokenData()
+      .then(res => this.setState(Object.assign(this.state, { tokenData: res })))
+      .catch(e => this.setState(Object.assign(this.state, {
+        tokenData: `error fetching token data: ${e.message}`
+      })))
+    this.loadMultiSigData()
+      .then(res => this.setState(Object.assign(this.state, { multiSigData: res })))
+      .catch(e => this.setState(Object.assign(this.state, {
+        multiSigData: `error fetching multi-sig data: ${e.message}`
+      })))
+  }
+
+  render() {
+    return (
+      <div className="dapp-info-wrapper">
+        <div className="container">
+          <div className="row">
+            <div className="col-md-6">
+              <h1>DApp Info</h1>
+              <h3 className="lead">Developer information about this build</h3>
+              <table width="100%">
+
+                <tr><th colSpan="2">origin.js</th></tr>
+                <tr><td>Version</td><td>{ origin.version }</td></tr>
+
+                <tr><th colSpan="2">Environment Variables</th></tr>
+                <tr><td>BRIDGE_SERVER_DOMAIN</td><td>{ process.env.BRIDGE_SERVER_DOMAIN }</td></tr>
+                <tr><td>BRIDGE_SERVER_PROTOCOL</td><td>{ process.env.BRIDGE_SERVER_PROTOCOL }</td></tr>
+                <tr><td>DISCOVERY_SERVER_URL</td><td>{ process.env.DISCOVERY_SERVER_URL }</td></tr>
+                <tr><td>IPFS_DOMAIN</td><td>{ process.env.IPFS_DOMAIN }</td></tr>
+                <tr><td>IPFS_GATEWAY_PORT</td><td>{ process.env.IPFS_GATEWAY_PORT }</td></tr>
+                <tr><td>IPFS_GATEWAY_PROTOCOL</td><td>{ process.env.IPFS_GATEWAY_PROTOCOL }</td></tr>
+                <tr><td>IPFS_API_PORT</td><td>{ process.env.IPFS_API_PORT }</td></tr>
+                <tr><td>IPFS_SWARM</td><td>{ process.env.IPFS_SWARM }</td></tr>
+                <tr><td>ARBITRATOR_ACCOUNT</td><td>{ process.env.ARBITRATOR_ACCOUNT }</td></tr>
```

Comment:
While we are at it, do you mind adding the AFFILIATE_ACCOUNT ?

Reply:
done